tower-http/src/trace.rs

```diff
@@ -0,0 +1,230 @@
+use crate::LatencyUnit;
+use crate::{GetTraceStatus, GetTraceStatusFromHttpStatus, TraceStatus};
+use futures_util::ready;
+use http::{Request, Response};
+use pin_project::pin_project;
+use std::future::Future;
+use std::time::Instant;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tower_layer::Layer;
+use tower_service::Service;
+use tracing::{
+    field::{debug, display},
+    Level, Span,
+};
+
+#[derive(Clone, Debug)]
+pub struct TraceLayer<T> {
+    record_headers: bool,
+    span: Option<Span>,
+    latency_unit: LatencyUnit,
+    get_trace_status: T,
+    record_full_uri: bool,
+}
+
+impl Default for TraceLayer<GetTraceStatusFromHttpStatus> {
```

Comment:
should this maybe be a default parameter on `TraceLayer` as well?

Reply:
Good idea!